### PR TITLE
Try using isoinfo(1) to extract an ISO image identifier

### PR DESCRIPTION
**Pull request by iskunk**
The first of several PRs stemming from #13. This uses a utility from the `genisoimage` package, if available, to extract some kind of title string for each ISO image file found.

### CODE REVIEW

File: 42_grml [hunk 1]
```diff
@@ -74,10 +74,18 @@ get_dependencies() {
     break
   done
   echo $dependencies
-
 }
 
-
+get_iso_identifier()
+{
+  local iso=${1}
```
Comment (mika) [inline 42_grml:81]: "local" sadly isn't strictly POSIX, but AFAICT it shouldn't be a problem for bash, dash,... or any relevant shells that might be present for /bin/sh on Debian or alike systems.

But please double quote the "$1" (and nitpick: I prefer to use $X for single char variables and ${foo} for multi char variables)

File: 42_grml [hunk 2]
```diff
@@ -74,10 +74,18 @@ get_dependencies() {
...
+  local id=
+  if isoinfo --version >/dev/null 2>&1 ; then
```
Comment (mika) [inline 42_grml:83]: Is there any specific reason why we invoke isoinfo with --version and not check for its availability/executable?
(No big deal though and fine for me, just curious!)

File: 42_grml [hunk 3]
```diff
@@ -74,10 +74,18 @@ get_dependencies() {
...
+    id=$(isoinfo -d -i $iso | grep '^Volume id:' | sed -e 's/^[^:]*: *//')
```
Comment (mika) [inline 42_grml:84]: Please quote $iso here as "${iso}" to prevent globbing/word splitting

File: 42_grml [hunk 4]
```diff
@@ -74,10 +74,18 @@ get_dependencies() {
...
+    [ -n "$id" ] || id=$(isoinfo -d -i $iso | grep '^Application id:' | sed -e 's/^[^:]*: *//')
```
Comment (mika) [inline 42_grml:85]: same :)

File: 42_grml [hunk 5]
```diff
@@ -74,10 +74,18 @@ get_dependencies() {
...
+  fi
+  echo "$id"
```
Comment (mika) [inline 42_grml:87]: nit: echo "${id}"

File: 42_grml [hunk 6]
```diff
@@ -111,7 +119,12 @@ for grmliso in $iso_list ; do
   esac
 
   echo "Found Grml ISO image: $grmliso" >&2
-  title="Grml Rescue System ($grml)"
+  iso_id=$(get_iso_identifier $grmliso)
+  if [ -n "$iso_id" ] ; then
```
Comment (mika) [inline 42_grml:123]: nit: "${iso_id}" :) (and same below please while at it!)

File: 42_grml [hunk 7]
```diff
@@ -74,10 +74,18 @@ get_dependencies() {
     break
   done
   echo $dependencies
-
 }
 
-
+get_iso_identifier()
+{
```
Comment (mika) [inline 42_grml:80]: nitpick: please let's keep style consistent and use:

```
get_iso_identified() {
```

(instead of having the "{" in the next line)

**Review by mika — COMMENTED** [review]
Not yet tested but only went briefly through the code, only some nitpicks from me side!
Looking forward to have that available, thanks! :+1:

File: 42_grml (hunk 1 above)
Comment (jkirk) [reply 42_grml:81]: (FTR, the `$..$` (without single backticks) in the comment above are rendered as mathematical expression, see: https://docs.github.com/en/get-started/writing-on-github/working-with-advanced-formatting/writing-mathematical-expressions. I just fixed it with `<span>$</span>` without putting them in single backticks.)

File: 42_grml (hunk 2 above)
Comment (iskunk) [reply 42_grml:82]: I figure it's better than `test -x` (requires hard-coding the binary location) or trying to use `which`...

File: 42_grml (hunk 7 above)
Comment (iskunk) [reply 42_grml]: Whoops, didn't catch that one.

**Comment by iskunk** [comment]
All right, I've updated the PR. The feedback and guidance are appreciated.

Please let me know if anything remains that's not up to snuff.

**Comment by mika** [comment]
Thanks also for this, just verified and LGTM, rebased against your PR 15 and just submitted. Thanks again! :)

**Pull request merged**